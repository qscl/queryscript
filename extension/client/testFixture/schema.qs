export type User {
    id int,
    org_id int,
    name string,
}

export type Event {
    user_id int,
    description string,
    ts string,
}

export let users [User] = load('data/users.json');
export let events [Event] = load('data/events.json');
