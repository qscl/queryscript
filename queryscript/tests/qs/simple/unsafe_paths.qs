import schema;

-- This query fails but returns a flaky error
-- unsafe select * from schema.users;

let bar = schema.users;
unsafe select * from bar;

let baz = 1;

-- NOTE: Without the `as` statement, this will have a weird alias, because we don't
-- really distinguish between the value and the alias in unsafe mode.
unsafe select baz as baz from bar;

-- This will have the correct alias
unsafe select baz as baz from bar;
