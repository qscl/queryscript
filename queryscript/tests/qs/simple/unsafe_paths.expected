{
    "compile_errors": [
        (
            None,
            RuntimeError {
                source: DuckDBError {
                    source: DuckDBFailure(
                        Error {
                            code: Unknown,
                            extended_code: 1,
                        },
                        Some(
                            "Catalog Error: Schema with name schema does not exist!\nLINE 1: SELECT * FROM \"schema\".\"users\" LIMIT 0\n                      ^",
                        ),
                    ),
                    backtrace: None,
                },
                loc: Unknown,
            },
        ),
    ],
    "decls": {
        "let bar": [{
        	id Int32,
        	org_id Int32,
        	name Utf8,
        	active Boolean,
        }],
        "let baz": Int64,
    },
    "queries": [
        Err(
            StringError {
                what: "Unknown type cannot exist at runtime (?unsafe expr?)",
                backtrace: None,
            },
        ),
        Ok(
            TypedValue {
                type_: List(
                    Record(
                        [
                            Field {
                                name: "id",
                                type_: Atom(
                                    Int32,
                                ),
                                nullable: true,
                            },
                            Field {
                                name: "org_id",
                                type_: Atom(
                                    Int32,
                                ),
                                nullable: true,
                            },
                            Field {
                                name: "name",
                                type_: Atom(
                                    Utf8,
                                ),
                                nullable: true,
                            },
                            Field {
                                name: "active",
                                type_: Atom(
                                    Boolean,
                                ),
                                nullable: true,
                            },
                        ],
                    ),
                ),
                value: "| id | org_id | name | active |\n|----|--------|------|--------|\n| 1  | 1      | Foo  | true   |\n| 2  | 1      | Bar  | false  |",
            },
        ),
        Ok(
            TypedValue {
                type_: List(
                    Record(
                        [
                            Field {
                                name: "baz",
                                type_: Atom(
                                    Int64,
                                ),
                                nullable: true,
                            },
                        ],
                    ),
                ),
                value: "| baz |\n|-----|\n| 1   |\n| 1   |",
            },
        ),
        Ok(
            TypedValue {
                type_: List(
                    Record(
                        [
                            Field {
                                name: "baz",
                                type_: Atom(
                                    Int64,
                                ),
                                nullable: true,
                            },
                        ],
                    ),
                ),
                value: "| baz |\n|-----|\n| 1   |\n| 1   |",
            },
        ),
    ],
}